labelme/widgets/canvas.py

```diff
@@ -478,13 +492,16 @@ def selectShapePoint(self, point, multiple_selection_mode):
                 if self.isVisible(shape) and shape.containsPoint(point):
                     self.calculateOffsets(shape, point)
                     self.setHiding()
-                    if multiple_selection_mode:
-                        if shape not in self.selectedShapes:
+                    if shape not in self.selectedShapes:
+                        if multiple_selection_mode:
                             self.selectionChanged.emit(
                                 self.selectedShapes + [shape]
                             )
+                        else:
+                            self.selectionChanged.emit([shape])
+                        self.wasSelected = False
                     else:
-                        self.selectionChanged.emit([shape])
+                        self.wasSelected = True
```

Comment:
Can't we move 
```
                    self.selectionChanged.emit(
                        [x for x in self.selectedShapes if x != self.hShape]
                    )
```
here?

Reply:
I'm not sure how it would be different whether shapes are unselected when the mouse is clicked or released.

Reply:
Done.